setup.py

```diff
@@ -98,7 +98,7 @@
     'jupyter_client',
     'ipython',
     'ipykernel', # bless IPython kernel for now
-    'prompt_toolkit>=2.0.0,<2.1.0',
+    'prompt_toolkit>=2.0.0,<3.1.0,!=3.0.0,!=3.0.1',
```

Comment:
Why include 2.0.0? Why not just` >=3.0.2`?

Reply:
Wouldn't that express a minimum requirement, to insist that the consumer have _at least_ 2.0.0 installed? I'm not sure why that would be an issue? (Of course, if it's not a real minimum requirement and prompt-toolkit would work with earlier versions, then I agree with you.)

Reply:
Retain maximum compatibility. If we force `>=3.0.2` then any downstream projects will be forced to upgrade, whereas currently they could still choose to remain on 2.x.

Reply:
And to clarify, the version requirements are copied verbatim from ipython which caused the conflict in the first place :)

https://github.com/ipython/ipython/blob/master/setup.py#L193

Reply:
Sounds good to me! Merging...